### PR TITLE
Feat/api integration

### PR DESCRIPTION
**Pull request by riteshsp2000**
# Your checklist for this pull request

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [ ] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.
- [ ] I have added necessary documentation (if appropriate)

## Description
Fetch API and display mails

💖Thank you!

## Post merge checklist

- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

File: dalchini/src/pages/Home/index.tsx [hunk 1]
```diff
@@ -14,14 +14,57 @@ import {
...
   useEffect(() => {
     setTimeout(() => setLoading(false), 5000);
+    const fetchData = async () => {
+      try {
+        const {
+          data: { m: mails }
+        } = await api.get('/', {
+          params: {
+            url: 'inbox.json'
+            // id: 13440
```
Comment (abhibhaw) [inline dalchini/src/pages/Home/index.tsx:53]: Faaltu comment

File: dalchini/src/pages/MailView/index.tsx [hunk 2]
```diff
@@ -26,50 +72,72 @@ const MailView: React.FC = () => {
...
+                            downloadURI(
+                              // eslint-disable-next-line max-len
+                              `https://mail.nitrkl.ac.in/home/~/?auth=qp&zauthtoken=${getQueryParam(
+                                'zauthtoken'
+                              )}&id=${pathname.split('/')[2]}&part=${partId}`,
+                              filename
+                            )
+                          // window.open(
+                          // eslint-disable-next-line max-len
+                          //   // `https://mail.nitrkl.ac.in/service/home/~/&auth=qp?zauthtoken=${getQueryParam(
```
Comment (abhibhaw) [inline dalchini/src/pages/MailView/index.tsx:124]: These too

**Review by abhibhaw — DISMISSED** [review]
lgtm

**Review by DesignrKnight — DISMISSED** [review]
Your first two commits in this PR contain a token. Ideally they should never come to GitHub. I would recommend you to squash all these commits into one as the commits in PR are always visible to someone who wants to expose you.

**Comment by abhibhaw** [comment]
> Your first two commits in this PR contain a token. Ideally they should never come to GitHub. I would recommend you to squash all these commits into one as the commits in PR are always visible to someone who wants to expose you.

Yep will squash and delete branch :D

**Review by abhibhaw — APPROVED** [review]

**Review by nayakastha — APPROVED** [review]

**Comment by DesignrKnight** [comment]
> > Your first two commits in this PR contain a token. Ideally they should never come to GitHub. I would recommend you to squash all these commits into one as the commits in PR are always visible to someone who wants to expose you.
> 
> Yep will squash and delete branch :D

Please feel free to justify why commits are still visible and accessible then
https://github.com/dscnitrourkela/project-elaichi/pull/151/commits/91d3019a9eaf7cc406e62cf8a224175558b98956

**Comment by abhibhaw** [comment]
> > > Your first two commits in this PR contain a token. Ideally they should never come to GitHub. I would recommend you to squash all these commits into one as the commits in PR are always visible to someone who wants to expose you.
> > 
> > 
> > Yep will squash and delete branch :D
> 
> Please feel free to justify why commits are still visible and accessible then [91d3019](https://github.com/dscnitrourkela/project-elaichi/commit/91d3019a9eaf7cc406e62cf8a224175558b98956)

Because it wasn't from a fork? 🤧

**Pull request merged**